manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml

```diff
@@ -301,8 +301,13 @@ data:
         {{- end }}
       {{- end }}
         securityContext:
-          allowPrivilegeEscalation: {{ .Values.global.proxy.privileged }}
-          privileged: {{ .Values.global.proxy.privileged }}
+        {{ if or (eq (annotation .ObjectMeta `sidecar.istio.io/interceptionMode` .ProxyConfig.InterceptionMode) `TPROXY`)  .Values.global.proxy.privileged }}
```

Comment:
Again, running as privileged shouldn't be necessary. Just find out what capabilities are missing. I'd be very surprised if `NET_ADMIN` and `NET_RAW` were not sufficient. Do we need to install kernel modules, or something like that?

Also, no capabilities or privileged should be required if Istio CNI is enabled.

Reply:
Not sufficient, it need to set route to local network. exec `sysctl -w net.ipv4.conf.eth0.route_localnet=1`

https://github.com/moby/moby/issues/4717

Reply:
I think this if clause should also check whether CNI is enabled

Reply:
How can do that?

Reply:
See a few lines below your change.

Reply:
Just curious why do we even need `istio-init` when istio-cni is enabled

Reply:
Fixed ptal